src/mode/dns.rs

```diff
@@ -157,20 +159,21 @@ impl Dns {
     }
 
     fn flush(&mut self, ts: Timestamp) {
-        if (ts - self.last) >= Duration::seconds(1) {
-            let mut rs = Vec::with_capacity(self.buffer.len());
-            swap(&mut self.buffer, &mut rs);
-
-            let timeout = Duration::milliseconds(10).unsigned_abs();
-            let len = rs.len();
-            match self.client.send(rs, timeout) {
-                Ok(..) => debug!("DNS batch sent: {}", len),
-                Err(e) => warn!("DNS queue full: {:?}", e),
-            };
-
-            self.asm.flush(ts);
-            self.last = ts;
+        if (ts - self.last) < Duration::seconds(1) || self.buffer.len() < MAX_BUFFER_LEN {
```

Comment:
Shouldn't this be &&, not ||?  We want to send the buffer if either time or space limit is exceeded, so return if we're still inside both limits?

Reply:
Yes indeed!